---
title: "R for Bioinformatic Analyses"
author: "Hannah Tavalire and Bill Cresko - University of Oregon"
date: "January 2019 - Cesky Krumlov"
output: revealjs::revealjs_presentation
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r}
library(knitr)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
```

# Lecture 1 - Using R for Biostatistical Analyses

# But first a beautiful chair

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='50%', fig.align='center'}
knitr::include_graphics("images/week_01.001.jpeg")
```

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.align='center'}
knitr::include_graphics("images/week_01.002.jpeg")
```

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.align='center'}
knitr::include_graphics("images/week_01.003.jpeg")
```

___________________

## Before we talk about how amazing `R` is...
- navigate into this directory: ~/workshop_materials/evomics_stats_2019/
- type 'git pull'

# Why use `R`?

## {}
- R is a _statistical_ programming language (derived from S)
- Superb data management & graphics capabilities
- You can write your own functions
- Powerful and flexible
- Runs on all computer platforms
- Well established system of packages and documentation
- Active development and dedicated community
- Can use a nice GUI front end such as `Rstudio`
- Reproducibility 
    - keep your scripts to see exactly what was done
    - distribute these with your data
    - embed your `R` analyses in polished `RMarkdown` files
- FREE
    
## {.flexbox .vcenter}
```{r, echo=FALSE, out.width='35%'}
knitr::include_graphics("images/volcano_plot.png")
knitr::include_graphics("images/rainbow_boxplot.png")
knitr::include_graphics("images/blob_plot.png")
knitr::include_graphics("images/Stacked_barplot.png")

```

## `R` resources  {.flexbox .vcenter}

- The R Project Homepage: http://www.r-project.org
- Quick R Homepage: http://www.statmethods.net
- Bioconductor: http://www.bioconductor.org    
- An Introduction to R (long!): http://cran.r-project.org/doc/manuals/R-intro.html
- R for Data Science: https://r4ds.had.co.nz
- Google - tutorials, guides, demos, packages and more

## Running `R` {.flexbox .vcenter}
- Need to make sure that you have `R` installed 
    - locally or on a server
    - https://www.r-project.org
  
- Run R from the command line
    - just type `R`
    - can run it locally as well as on clusters

- Install a R *Integrated Development Environment* (IDE)
    - RStudio: http://www.rstudio.com
    - Makes working with R much easier, particularly for a new R user
    - Run on Windows, Mac or Linux OS
    - We're running as a server on the AWS instances

## `RStudio` {.flexbox .vcenter}

```{r, echo=FALSE, out.width='75%', fig.align='center'}
knitr::include_graphics("images/RStudio_Screenshot.png")
```


## Exercise 1.1 - Exploring `RStudio` {}
- Open `RStudio` by adding `:8787` to your AMI url   
- Take a few minutes to familiarize yourself with the `Rstudio` environment by locating the following features:
    - See what types of new files can be made in `Rstudio` by clicking the top left icon- open a new R script.
    - The windows clockwise from top left are: the code editor, the workspace and history, the plots and files window, and the R console.
    - In the plots and files window, click on the packages and help tabs to see what they offer. 
- Now open the file called *Exercises_for_R_Lectures.Rmd* in /workshop_materials/evomics_stat_2019/03.Exercises/ 
    - This file will serve as your digital notebook for parts of the workshop and contains the other exercises.

___________________

# Introduction to `RMarkdown`

## `RMarkdown` {.flexbox .vcenter}

- A great way to embed `R code` into descriptive files to keep your life organized
    - http://kbroman.org/knitr_knutshell/pages/Rmarkdown.html
    - https://bookdown.org/yihui/rmarkdown/
    - https://bookdown.org/yihui/bookdown/
- You can insert `R chunks` into `Rmarkdown` documents
- You will be doing more with markdown on Thursday

## Exercise 1.2 - Intro to `RMarkdown` Files

- Take a few minutes to familiarize yourself with RMarkdown files by completing exercise 1.2 in your exercises document.

___________________

# BASICS of `R` 

## BASICS of `R` {}

- Commands can be submitted through 
    - terminal, console or scripts
    - can be embedded as `code chunks` in RMarkdown
- On these slides evaluating code chunks and showing output
    - shown here after the two `#` symbols
    - the number of output items is in `[]`
- `R` follows the normal priority of mathematical evaluation (PEDMAS)

## BASICS of `R` {}

Input code chunk and then output
```{r basic multiplication, echo=TRUE}
4*4
```

Input code chunk and then output
```{r more multiplication, echo=TRUE}
(4+3*2^2)
```

## Assigning Variables {.flexbox .vcenter}

- A better way to do this is to assign variables
- Variables are assigned values using the `<-` operator.
- Variable names must begin with a letter, but other than that, just about anything goes.
- Do keep in mind that `R` is **case sensitive**.

## Assigning Variables {.flexbox .vcenter}

```{r assigning variables, echo=TRUE, tidy=TRUE}

x <- 2
x*3
y <- x * 3
y-2

```

These do not work

```{r eval=FALSE, echo=TRUE}
3y <- 3
3*y <- 3
```


## Arithmetic operations on functions {}

- Arithmetic operations can be performed easily on functions as well as numbers.

```{r functions, eval = TRUE, echo = TRUE}
x <- 12
x+2
x^2
log(x)
```

## Arithmetic operations on functions {}
- Note that the last of these - `log` - is a built in function of `R`, and therefore the object of the function needs to be put in parentheses
- These parentheses will be important, and we'll come back to them later when we add arguments after the object in the parentheses  
- The outcome of calculations can be assigned to new variables as well, and the results can be checked using the `print` command


## Arithmetic operations on functions {.flexbox .vcenter}

```{r, echo=TRUE}
y <- 67
print(y)

x <- 124
z <- (x*y)^2
print(z)
```

## STRINGS {.flexbox .vcenter}
- Operations can be performed on _character_ variables as well
- Note that "characters" need to be set off by quotation marks to differentiate them from numbers
- The `c` stands for `concatenate`
- Note that we are using the same variable names as we did previously, which means that we're overwriting our previous assignment
- A good rule of thumb is to use new names for each variable, and make them short but still descriptive

## STRINGS {.flexbox .vcenter}

```{r, echo=TRUE}
x <- "I Love"
print (x)
y <- "Biostatistics"
print (y)
z <- c(x,y)
print (z)
```


## VECTORS {.flexbox .vcenter}

- In general `R` thinks in terms of vectors
    - a list of characters, factors or numerical values ("I Love")
    - it will benefit any `R` user to try to write scripts with that in mind
    - it will simplify most things
- Vectors can be assigned directly using the 'c()' function and then entering the exact values.

## VECTORS {.flexbox .vcenter}

```{r Example vectors, echo=TRUE}
n <- c(2,3,4,2,1,2,4,5,10,8,9)
print(n)
```

```{r Example vectors two, echo=TRUE}
z <- n +3
print(z)
```

## FACTORS {.flexbox .vcenter}

- The vector `x` is now what is called a list of character values ("I Love").
- Sometimes we would like to treat the characters as if they were units for subsequent calculations.
- These are called `factors`, and we can redefine our character variables as factors.
- This might seem a bit strange, but it’s important for statistical analyses where we might want to see the mean or variance for two different treatments.

## FACTORS {.flexbox .vcenter}

```{r, eval = TRUE, echo = TRUE}
x_factor <- as.factor(x)
print (x_factor)
```
- Note that factor levels are reported alphabetically

## FACTORS {.flexbox .vcenter}

- We can also determine how `R` "sees" a variable using `str()` or `class()` functions. 
- This is a useful check when importing datasets or verifying that you assigned a class correctly
```{r, eval = TRUE, echo = TRUE}
str(x)
class(x)
```


## Types or 'classes' of vectors of data {.smaller}

```{r, echo=FALSE, out.width='50%', fig.align='center'}
knitr::include_graphics("images/week_3.014.jpeg")
```

## Types of vectors of data {}

- `int` stands for integers  

- `dbl` stands for doubles, or real numbers

- `chr` stands for character vectors, or strings  

- `dttm` stands for date-times (a date + a time)  

- `lgl` stands for logical, vectors that contain only TRUE or FALSE

- `fctr` stands for factors, which R uses to represent categorical variables with fixed possible values  

- `date` stands for dates  

## Types of vectors of data {.smaller}

- Logical vectors can take only three possible values:
    - `FALSE`
    - `TRUE`
    - `NA` which is 'not available'.

- Integer and double vectors are known collectively as numeric vectors.
    - In `R` numbers are doubles by default.

- Integers have one special value: NA, while doubles have four:
    - `NA`
    - `NaN` which is 'not a number'
    - `Inf`
    - `-Inf`

## Basic Statistics {.flexbox .vcenter}

Many functions exist to operate on vectors.
```{r, eval = FALSE, echo = TRUE}
mean(n)
median(n)
var(n)
log(n)
exp(n)
sqrt(n)
sum(n)
length(n)
sample(n, replace = T) #has an additional argument (replace=T)
```
- Arguments modify or direct the function in some way
    - There are many arguments for each function, some of which are defaults
    - Tab complete is helpful to view argument options

## Getting Help {.flexbox .vcenter}

- Getting Help on any function is very easy - just type a question mark and the name of the function.
- There are functions for just about anything within `R` and it is easy enough to write your own functions if none already exist to do what you want to do.
- In general, function calls have a simple structure: a function name, a set of parentheses and an optional set of parameters/arguments to send to the function.
- Help pages exist for all functions that, at a minimum, explain what parameters exist for the function.  


## Getting Help {.flexbox .vcenter}

```{r, eval = FALSE, echo = TRUE}
- help(mean)
- ?mean
- example(mean)
- help.search("mean")
- apropos("mean")
- args(mean)
```

## Creating vectors {}

- Creating a vector of new data by entering it by hand can be a drag
- However, it is also very easy to use functions such as
    - `seq`
    - `sample`

## Creating vectors {}
- What do the arguments mean?
```{r, echo = TRUE}
seq_1 <- seq(0.0, 10.0, by = 0.1)
print(seq_1)
```

## Creating vectors {}
```{r, echo = TRUE}
seq_2 <- seq(10.0, 0.0, by = -0.1)
print(seq_2)
```

## Creating vectors {.smaller}

```{r, echo = TRUE}
seq_square <- (seq_2)*(seq_2)
print(seq_square)
```

## Creating vectors {.smaller}

```{r, echo = TRUE}
seq_square_new <- (seq_2)^2
print(seq_square_new)
```

## R Interlude 

Complete Exercises 1.3-1.6

```{r, echo=FALSE, out.width='50%', fig.align='center'}
knitr::include_graphics("images/Mr_Rog_and_cat.png")
```


## Drawing samples from distributions {.smaller}

- Here is a way to create your own data sets that are random samples...
```{r Samples from distributions 2, out.width='50%', echo = TRUE}
x <- rnorm (n = 10000, mean = 0, sd = 10)
y <- sample (1:10000, 10000, replace = T)
xy <- cbind(x,y)
plot(xy)
```

## Drawing samples from distributions {.smaller}

```{r Samples from distributions 3, out.width='50%', echo = TRUE}
x <- rnorm (10000, 0, 10)
y <- sample (1:10000, 10000, replace = T)
xy <- cbind(x,y)
hist(x)
```

## Drawing samples from distributions {.flexbox .vcenter}

- You’ve probably figured out that y from the last example is drawing numbers with equal probability.
- What if you want to draw from a distribution?
- Again, play around with the arguments in the parentheses to see what happens.

```{r Samples from distributions 4, out.width='50%', echo = TRUE, eval=F}
x <- rnorm (10000, 0, 10)
y <- sample (???, 10000, replace = ???)
```

## Drawing samples from distributions {.flexbox .vcenter out.width='50%'}

- `dnorm()` generates the probability density, which can be plotted using the `curve()` function.
- Note that is curve is added to the plot using `add=TRUE`

```{r, out.width='30%', echo = TRUE}
x <-rnorm(1000, 0, 100)
hist(x, xlim = c(-500,500))
curve(50000*dnorm(x, 0, 100), xlim = c(-500,500), add=TRUE, col='Red')
```

# Visualizing Data in `R`

## Visualizing Data {.flexbox .vcenter}

- So far you've been visualizing just the list of output numbers
- Except for the last example where I snuck in a `hist` function.
- You can also visualize all of the variables that you've created using the `plot` function (as well as a number of more sophisticated plotting functions).
- Each of these is called a `high level` plotting function, which sets the stage
- `Low level` plotting functions will tweak the plots and make them beautiful

## Visualizing Data {.flexbox .vcenter .smaller}

```{r, out.width='50%', echo = TRUE}
seq_1 <- seq(0.0, 10.0, by = 0.1)
plot (seq_1, xlab="space", ylab ="function of space", type = "p", col = "red")
```

## Putting plots in a single figure {.flexbox .vcenter}
- The first line of the lower script tells R that you are going to create a composite figure that has two rows and two columns (on next slide)
    - Can you tell how?
```{r, echo=TRUE, outwidth="40%", out.height="20%"}
seq_1 <- seq(0.0, 10.0, by = 0.1)
seq_2 <- seq(10.0, 0.0, by = -0.1)
```
```{r, echo=TRUE, eval=FALSE, outwidth="50%", out.height="40%"}
par(mfrow=c(2,2))
plot (seq_1, xlab="time", ylab ="p in population 1", type = "p", col = 'red')
plot (seq_2, xlab="time", ylab ="p in population 2", type = "p", col = 'green')
plot (seq_square, xlab="time", ylab ="p2 in population 2", type = "p", col = 'blue')
plot (seq_square_new, xlab="time", ylab ="p in population 1", type = "l", col = 'yellow')
```

## Putting plots in a single figure {.flexbox .vcenter .smaller}

```{r, echo=FALSE, eval=TRUE, outwidth="50%", out.height="50%"}
par(mfrow=c(2,2))
plot (seq_1, xlab="time", ylab ="p in population 1", type = "p", col = 'red')
plot (seq_2, xlab="time", ylab ="p in population 2", type = "p", col = 'green')
plot (seq_square, xlab="time", ylab ="p2 in population 2", type = "p", col = 'blue')
plot (seq_square_new, xlab="time", ylab ="p in population 1", type = "l", col = 'yellow')
```

## R Interlude 

Complete Exercises 1.7-1.8


# Working with Imported Datasets in `R`

## Creating Data Frames in `R` {.smaller}

- As you have seen, in `R` you can generate your own random data set drawn from nearly any distribution very easily.
- Often we will want to use collected data.
- Now, let’s make a dummy dataset to get used to dealing with data frames
    - Set up three variables (habitat, temp and elevation) as vectors

```{r, echo=TRUE}
habitat <- factor(c("mixed", "wet", "wet", "wet", "dry", "dry", "dry","mixed"))
temp <- c(3.4, 3.4, 8.4, 3, 5.6, 8.1, 8.3, 4.5)
elevation <- c(0, 9.2, 3.8, 5, 5.6, 4.1, 7.1, 5.3)
```

## Creating Data Frames in R {.smaller}
- Create a data frame where vectors become columns

```{r, echo=TRUE}
mydata <- data.frame(habitat, temp, elevation)
row.names(mydata) <- c("Reedy Lake", "Pearcadale", "Warneet", "Cranbourne", "Lysterfield", "Red Hill", "Devilbend", "Olinda")
head(mydata)
```
- Now you have a hand-made data frame with row names

## R Interlude: Reading in Data Frames in R
- A strength of `R` is being able to import data from an external source
    - Create the same table that you did above in a spreadsheet using LibreOffice 
    - Export it to comma separated and tab separated text files for importing into `R`.
    - The first will read in a comma-delimited file, whereas the second is a tab-delimited
    - In both cases the header and row.names arguments indicate that there is a header row and row label column
    - Note that the name of the file by itself will have `R` look in the PWD, whereas a full path can also be used

## Reading in Data Frames in R {.flexbox .vcenter}

```{r, eval = FALSE, echo = TRUE}
YourFile <- read.table('yourfile.csv', header=T, row.names=1, sep=',')
YourFile <- read.csv('yourfile.csv', header=T, row.names=1, sep=',')
YourFile <- read.table('yourfile.txt', header=T, row.names=1, sep='\t')
```

## Exporting Data Frames in R {.flexbox .vcenter}

```{r, eval = FALSE, echo = TRUE}
write.csv(YourFile, "yourfile.csv", quote=F, row.names=T, sep=",")
write.table(YourFile, "yourfile.txt", quote=F, row.names=T, sep="\t")
```
- you will get more practice with this during the next R interlude

# Where we left off...

## Where we left off...
- Use `:8787` to access `R studio` again 
- Please rename your _Exercises_ document before doing any `git pull`s today, as to not overwrite your work!
    - You can do this through the terminal window _in R studio_ using the `mv` command
- Working with imported datasets and reading and writing datasets
- Next up: indexing!
- _But first_- a note about arguments...

## Arguments in `R` Functions

- Sometimes `R` can guess what you mean because of order...

```{r, eval=T, echo=T}
x <- rnorm(1000, 0, 10)  #n=, mean=, sd=
x[1:10]
```

- But sometimes if the order isn't right, you can confuse `R` and get something you _really_ didn't want...
```{r, eval=T, echo=T}
x2 <- rnorm(10, 1000, 0)  #n=, mean=, sd=
x2
```

## Arguments in `R` Functions

- A work-around and best-practice: include the arguments!!
```{r, eval=T, echo=T}
set.seed(145)
x <- rnorm(n=1000, mean=0, sd=10)  #n=, mean=, sd=
x[1:10]
```

```{r, eval=T, echo=T}
set.seed(145)
x2 <- rnorm(sd=10, n=1000, mean=0)  #n=, mean=, sd=
x2[1:10]
```
   
  - Notice we also set the seed to replicate our sample results!


## Indexing in data frames {.flexbox .vcenter}
- Next up - indexing just a subset of the data
- This is a very important feature in `R`, that allows you to analyze just a subset of the data.
```{r, echo=TRUE, eval=FALSE}
print (YourFile[,2])
print (YourFile$temp)
print (YourFile[2,])
plot (YourFile$temp, YourFile$elevation)
```

## Indexing in data frames {.flexbox .vcenter}
- You can also assign values, or single values, from a data set to a new variable

```{r, echo=TRUE, eval=FALSE}
x <- (YourFile[,2])
y <-  (YourFile$temp)
z <- (YourFile$elevation)
plot (y, z)
```

## Indexing in data frames {.flexbox .vcenter}
- You can perform operations on particular levels of a factor
- Note that the first argument is the numerical column vector, and the second is the factor column vector.
- The third is the operation. **Reversing the first two does not work**
    - Tab complete will tell you the correct order for arguments 

```{r, eval = FALSE, echo = TRUE}
tapply(YourFile$temp, YourFile$habitat, mean)
tapply(YourFile$temp, YourFile$habitat, var)
```

## R Interlude

Complete Exercises 1.9-1.10


________________________

# Lecture 2 - Collaboration, Documentation and Reproducibility

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='50%', fig.align='center'}
knitr::include_graphics("images/week_01.001.jpeg")
```

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.align='center'}
knitr::include_graphics("images/Messy-Lab.jpg")
```

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.align='center'}
knitr::include_graphics("images/example_screenshot_labnotebook.png")
```

________________________

# Collaboration using Git and GitHub

## Git and GitHub {.vcenter .flexbox}

https://learngitbranching.js.org/

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.007.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.008.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.009.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.010.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.011.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.012.jpeg")
```

-------

```{r, echo=FALSE, out.width='90%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_5a.013.jpeg")
```


## Clone the repository

- First make a *new directory* into which you will clone our course repository
    - This will prevent you from overwriting any of the documents you have have edited
    - And it's good practice to do it again
- You should work through the terminal application and use Unix to do this
- Open the terminal and navigate to your new directory and type the following:

```{bash, echo=TRUE, eval=FALSE}
git clone https://github.com/wcresko/evomics_stat_2019.git
````

## Update the repository

- Now to update the repository you just need to use these commands

```{bash, echo=TRUE, eval=FALSE}
git status

git merge origin/master
```
- The first command just tells you if anything has changed  
- If so, do the second!
- This is **much safer** than `git pull` 

## Git and GitHub Interlude: Exercise 2.1
- Please read the directions carefully to prevent pushing to the wrong repository

________________________

# Clarity using Markdown and LaTeX

## What is markdown?  {}
- Lightweight _formal_ markup languages are used to add formatting to plaintext documents
    - Adding basic syntax to the text will make elements look different once rendered/knit
    - Available in many base editors (e.g., Atom text editor)
- You then need a markdown application with a markdown processor/parser to render your text files into something more exciting
    - Static and dynamic outputs!
    - pdf, HTML, presentations, websites, scientific articles, books etc
    
## What is Knitr and PANDOC?  {}
- Knitr is a package in R to render markdown files
- PANDOC is a general way to render markdown files into something else
- https://pandoc.orgis
- Can include math using LaTeX
- GitHub will render markdown directly
- Markdown can easily be rendered within most editors now
- Within RStudio just use the `knit` button to render markdown
- Markdown syntax is very easy
    
## Formatting text  {}    

```{r, eval=FALSE, echo=TRUE}
*Italic* or _Italic_
**Bold** or __Bold__
```

- *Italic* or _Italic_
- **Bold** or __Bold__


## Formatting text  {}  

```{r, eval=FALSE, echo=TRUE}
> "You know the greatest danger facing us is ourselves, an irrational fear of the unknown. 
But there’s no such thing as the unknown — only things temporarily hidden, temporarily not understood."
>
> --- Captain James T. Kirk
```

> "You know the greatest danger facing us is ourselves, an irrational fear of the unknown. But there’s no such thing as the unknown — only things temporarily hidden, temporarily not understood."
>
> --- Captain James T. Kirk

## Formatting lists  {}    

```{r, eval=FALSE, echo=TRUE}
- list_element
    - sub_list_element #double tab to indent
    - sub_list_element #double tab to indent
    - sub_list_element #double tab to indent
- list_element
    - sub_list_element #double tab to indent
#note the space after each dash- this is important!
```

- list_element
    - sub_list_element
    - sub_list_element
    - sub_list_element
- list_element
    - sub_list_element
  

## Formatting lists  {}    

```{r, eval=FALSE, echo=TRUE}
1. One
2. Two
3. Three
4. Four
```

1. One
2. Two
3. Three
4. Four

## Inserting images or URLs  {}    

```{r, eval=FALSE, echo=TRUE}
[Link](https://commonmark.org/help/)
![Image](https://i1.wp.com/evomics.org/wp-content/uploads/2012/07/20120115-IMG_0297.jpg)
```

[Link](https://commonmark.org/help/)
![Image](https://i1.wp.com/evomics.org/wp-content/uploads/2012/07/20120115-IMG_0297.jpg)

## What is LaTeX?  {.flexbox .vcenter}

- Pronounced «Lah-tech» or «Lay-tech» (to rhyme with «Bertolt Brecht»)
- A document preparation system for high-quality typesetting 
- It is most often used for medium-to-large technical or scientific documents
- Can be used for almost any form of publishing.
- Typesetting journal articles, technical reports, books, and slide presentations
- Allows for precise mathematical statements
- https://www.latex-project.org

## What is LaTeX?  {}

- LaTeX is not a word processor! 
- LaTeX encourages authors not to worry too much about the appearance of their documents but to concentrate on getting the right content. 
- Control over large documents containing sectioning, cross-references, tables and figures.
- Typesetting of complex mathematical formulas.
- Automatic generation of bibliographies and indexes.
- Multi-lingual typesetting
- https://bookdown.org/yihui/bookdown/


## What is LaTeX?  {}

- **Importantly, LaTeX can be included right into RMarkdown documents**
- The following slides have some examples

## Operators and Symbols

```{r, eval=FALSE, echo=TRUE}
$$ \large a^x, \sqrt[n]{x}, \vec{\jmath}, \tilde{\imath}$$
```

$$ \large a^x, \sqrt[n]{x}, \vec{\jmath}, \tilde{\imath}$$

```{r, eval=FALSE, echo=TRUE}
$$ \large \alpha, \beta, \gamma$$
```
$$ \large \alpha, \beta, \gamma$$

## Operators and Symbols

```{r, eval=FALSE, echo=TRUE}
$$ \large\approx, \neq, \nsim $$
```
$$ \large\approx, \neq, \nsim $$

```{r, eval=FALSE, echo=TRUE}
$$\large \partial, \mathbb{R}, \flat$$
```
$$\large \partial, \mathbb{R}, \flat$$

## Equations

Binomial sampling equation

```{r, eval=FALSE, echo=TRUE}
$$\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}$$
```

$$\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}$$

Poisson Sampling Equation

```{r, eval=FALSE, echo=TRUE}
$$\large Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}$$
```

$$\large Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}$$



## Differential Equations

```{r, eval=FALSE, echo=TRUE}
$$\iint xy^2\,dx\,dy =\frac{1}{6}x^2y^3$$
```
$$\iint xy^2\,dx\,dy =\frac{1}{6}x^2y^3$$

## Matrix formulations

```{r, eval=FALSE, echo=TRUE}
$$	\begin{matrix}
		-2 & 1 & 0 & 0 & \cdots & 0  \\
		1 & -2 & 1 & 0 & \cdots & 0  \\
		0 & 1 & -2 & 1 & \cdots & 0  \\
		0 & 0 & 1 & -2 & \ddots & \vdots \\
		\vdots & \vdots & \vdots & \ddots & \ddots & 1  \\
		0 & 0 & 0 & \cdots & 1 & -2
	\end{matrix} $$
```
	
$$	\begin{matrix}
		-2 & 1 & 0 & 0 & \cdots & 0  \\
		1 & -2 & 1 & 0 & \cdots & 0  \\
		0 & 1 & -2 & 1 & \cdots & 0  \\
		0 & 0 & 1 & -2 & \ddots & \vdots \\
		\vdots & \vdots & \vdots & \ddots & \ddots & 1  \\
		0 & 0 & 0 & \cdots & 1 & -2
	\end{matrix} $$
	
## Including LaTeX and Code into Markdown Files {.flexbox .vcenter}

- Explicit inclusion of code and mathematical equations helps with reproducibility
- Need to designate the 'environment' as being code or math
- Can be included in-line or in 'chunks'

## In-line versus fenced

```{r, eval=FALSE, echo=TRUE}
This equation, $y=\frac{1}{2}$, is included inline
```

This equation, $y=\frac{1}{2}$, is included inline

```{r, eval=FALSE, echo=TRUE}
Whereas this equation, $$y=\frac{1}{2}$$, is put on a separate line
```

Whereas this equation $$y=\frac{1}{2}$$ is put on a separate line

## Markdown is very flexible
- You can import RMarkdown templates into RStudio and open as a new Rmarkdown file
- Better yet there are packages that add functionality
- When you install the package it will show up in the 'From Template' section of the 'new file' startup screen
- There are packages to make
    - books
    - journal articles
    - slide shows
    - interactive exercises
    - many more
- Some of these use 'Shiny'
    - an interactive web based application
    - allows users to input and get output

## Final Thoughts on Markdown, LaTeX and GitHub
- Many forms/flavors of markdown
    - HTML and Rmarkdown are just forms of markdown
    - There is a GitHub flavored markdown
    - Once you learn one, all the others are very easy
- The goal is increased collaboration and reproducibility
    - Allows you to easily work with others by sharing the markdown file
    - Allows **formal** representation of code and math
    - Allows others to run your code directly
    - Allows reports to nontech people
    - All files are easily shared on GitHub
- Once you start using Markdown you won't stop.....

## R Interlude | Exploring RMarkdown 
- Exercise 2.2

________________________

# Data wrangling and exploratory data analysis (EDA)

## A biological example to get us started

Say you perform an experiment on two different strains of stickleback fish, one from an ocean population (RS) and one from a freshwater lake (BP) by making them microbe free. Microbes in the gut are known to interact with the gut epithelium in ways that lead to a proper maturation of the immune system.

## A biological example to get us started

You carry out an experiment by treating multiple fish from each strain so that some of them have a conventional microbiota, and some are inoculated with only one bacterial species. You then measure the levels of gene expression in the stickleback gut using RNA-seq. You suspect that the sex of the fish might be important so you track it too.

## A biological example to get us started

```{r, echo=FALSE, out.width='50%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/week_01.005.jpeg")
```

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/week_01.006.jpeg")
```

## Collecting Data with Analyses in Mind

- How should the data set be organized to best analyze it?
- What are the key properties of the variables?
- Why does that matter for learning `R`?
- Why does that matter for performing statistical analyses?


## Data set rules of thumb (aka Tidy Data)

- Store a copy of data in non-proprietary formats
- Leave an uncorrected file when doing analyses
- Maintain effective metadata about the data
- When you add observations to a database, add rows
- When you add variables to a database, add columns
- A column of data should contain only one data type

## {.flexbox .vcenter}

```{r, echo=FALSE, out.width='100%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/week_01.007.jpeg")
```

____________________________________________________

# Tidyverse family of packages {.smaller}

## Tidyverse family of packages {.smaller}

```{r, echo=FALSE, out.width='90%', fig.align='center'}
knitr::include_graphics("images/week_3.011.jpeg")
```

## Tidyverse family of packages {.vcenter .flexbox}

- Hadley Wickham and others have written `R` packages to modify data

- These packages do many of the same things as base functions in `R`

- However, they are specifically designed to do them faster and more easily

- Wickham also wrote the package `GGPlot2` for elegant graphics creations

- GG stands for ‘Grammar of Graphics’


## Example of a tibble {.smaller}

```{r, echo=FALSE, out.width='80%', fig.align='center'}
knitr::include_graphics("images/week_3.012.jpeg")
```

## Example of a tibble {.smaller}

```{r, echo=FALSE, out.width='80%', fig.align='center'}
knitr::include_graphics("images/week_3.013.jpeg")
```

## Key functions in `dplyr` for vectors

- Pick observations by their values with `filter()`.
- Reorder the rows with `arrange()`.
- Pick variables by their names with `select()`.
- Create new variables with functions of existing variables with `mutate()`.
- Collapse many values down to a single summary with `summarise()`.

## `filter()`, `arrange()` & `select()`

```{r, echo=T, eval=F}
filter(flights, month == 11 | month == 12)
```

```{r, echo=T, eval=F}
arrange(flights, year, month, day)
```

```{r, echo=T, eval=F}
select(flights, year, month, day)
```

```{r, echo=FALSE, out.width='60%', fig.align='center'}
knitr::include_graphics("images/week_3.015.jpeg")
```


## `mutate()` & `transmutate()`

This function will add a new variable that is a function of other variable(s)

```{r, echo=T, eval=F}
mutate(flights,
  gain = arr_delay - dep_delay,
  hours = air_time / 60,
  gain_per_hour = gain / hours
)
```

This function will replace the old variable with the new variable

```{r, echo=T, eval=F}
transmute(flights,
  gain = arr_delay - dep_delay,
  hours = air_time / 60,
  gain_per_hour = gain / hours
)
```

## `group_by( )` & `summarize( )`

This first function allows you to aggregate data by values of categorical variables (factors)

```{r, echo=T, eval=F}
by_day <- group_by(flights, year, month, day)
```

Once you have done this aggregation, you can then calculate values (in this case the mean) of other variables split by the new aggregated levels of the categorical variable

```{r, echo=T, eval=F}
summarise(by_day, delay = mean(dep_delay, na.rm = TRUE))
```

## `group_by( )` & `summarize( )`

- Note - you can get a lot of missing values!
- That’s because aggregation functions obey the usual rule of missing values:
    - if there’s any missing value in the input, the output will be a missing value.
    - fortunately, all aggregation functions have an na.rm argument which removes the missing values prior to computation

## R INTERLUDE | Complete Exercise 2.3-2.4 {.flexbox .vcenter}

_________________________________

# Graphical Communication

## GGPlot2 and the Grammar of Graphics

- GG stands for ‘Grammar of Graphics’
- A good paragraph uses good grammar to convey information
- A good figure uses good grammar in the same way
- Seven general components can be used to create most figures

## GGPlot2 and the Grammar of Graphics

```{r, echo=FALSE, fig.cap="xxx", out.width='80%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.017.jpeg")
```

## Graphical representation | general approaches {.smaller}

1. Distributions of data
    * location
    * spread
    * shape
  
2. Associations between variables
    * relationship among two or more variables
    * differences among groups in their distributions
  
## Graphical representation | general approaches {.smaller}

1. Distributions of data
    * bar graph
    * histogram
    * box plot
  
2. Associations between variables
    * pie chart
    * grouped bar graph
    * mosaic plot
    * box plot
    * scatter plot
    * dot plot 'stripchart'

## Box Plot {.flexbox .vcenter .smaller}

- Displays median, first and third quartile, range, and extreme observations 
- Can be combined with mean and standard error of the mean
- Concise way to visualize many aspects of distribution

```{r, echo=FALSE, out.width='50%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.001.jpeg")
```

## Scatter Plot {.flexbox .vcenter .smaller}

- Displays association between two numerical variables 
- Goal is association not magnitude or frequency  
- Points fill the space available 

```{r, echo=FALSE, out.width='50%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.002.jpeg")
```


## Examples of the good, bad and the ugly of graphical representation {.flexbox .vcenter}

- Examples of bad graphs and how to improve them.
- Courtesy of K.W. Broman  
- www.biostat.wisc.edu/~kbroman/topten_worstgraphs/ 

## Ticker tape parade {.flexbox .vcenter .smaller}

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.003.jpeg")
```

## A line to no understanding {.flexbox .vcenter .smaller}

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.004.jpeg")
```

## A cup of hot nothing {.flexbox .vcenter .smaller}

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.006.jpeg")
```

## A bake sale of pie charts {.flexbox .vcenter .smaller}

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.007.jpeg")
```

## Wack a mole

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.008.jpeg")
```

# Graphical communication best practices

##

> "Graphical excellence is that which gives to the viewer the greatest number of ideas in the shortest time  with the least ink in the smallest space"
>
> --- Edward Tufte

## Principles of effective display

- Show the data  
- Encourage the eye to compare differences 
- Represent magnitudes honestly and accurately  
- Draw graphical elements clearly, minimizing clutter  
- Make displays easy to interpret 

## “Above all else show the data” | Tufte  1983 

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.010.jpeg")
```

## “Maximize  the  data to ink  ratio,  within  reason” | Tufte   1983   

Draw  graphical  elements  clearly,  minimizing  clutter  

```{r, echo=FALSE, out.width='50%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.011.jpeg")
```


## “A  graphic  does  not  distort  if  the  visual  representation  of  the  data  is  consistent  with  the  numerical   representation” – Tufte   1983   

Represent  magnitudes  honestly  and  accurately  

```{r, echo=FALSE, out.width='60%', fig.asp=.6, fig.align='center'}
knitr::include_graphics("images/images_4a.012.jpeg")
```

## How Fox News makes a figure ...    

```{r, echo=FALSE, out.width='75%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.013.jpeg")
```

## How Fox News makes a figure ...   

```{r, echo=FALSE, out.width='30%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.013.jpeg")
```
```{r, echo=FALSE, out.width='30%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.014.jpeg")
```

“Graphical  excellence  begins  with  telling  the  truth  about  the  data”  –  Tufte    1983  

# Using GGPlot2 to make nice figures

## GGPlot2 and the Grammar of Graphics

```{r, echo=FALSE, fig.cap="xxx", out.width='80%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.017.jpeg")
```


## The `geom_bar` function {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
library(ggplot2)
ggplot(data=diamonds) +
  geom_bar(mapping=aes(x=cut))
```

## The `geom_bar` function {.flexbox .vcenter}
Now try this...

```{r gp1, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=diamonds) +
  geom_bar(mapping=aes(x=cut, colour=cut))
```

## The `geom_bar` function {.flexbox .vcenter}
and this...

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=diamonds) +
  geom_bar(mapping=aes(x=cut, fill=cut))
```

## The `geom_bar` function {.flexbox .vcenter}
and finally this...

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=diamonds) +
  geom_bar(mapping=aes(x=cut, fill=clarity), position="dodge")
```

## The `geom_histogram` and `geom_freqpoly` function {.flexbox .vcenter }
With this function you can make a histogram
```{r, eval=TRUE, echo=TRUE,fig.asp=0.4}
ggplot(data=diamonds) +
  geom_histogram(mapping=aes(x=carat), binwidth=0.5)
```

## The `geom_histogram` and `geom_freqpoly` function {.flexbox .vcenter }
This allows you to make a frequency polygram
```{r, eval=TRUE, echo=TRUE,fig.asp=0.4}
ggplot(data=diamonds) +
  geom_freqpoly(mapping=aes(x=carat), binwidth=0.5)
```

## The `geom_boxplot` function {.flexbox .vcenter}

Boxplots are very useful for visualizing data

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=diamonds, mapping=aes(x=cut, y=price)) +
  geom_boxplot()
```

## The `geom_boxplot` function {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg, mapping=aes(x=class, y=hwy)) +
  geom_boxplot() +
  coord_flip()
```

## The `geom_boxplot` function {.flexbox .vcenter}

```{r gp2, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg, mapping=aes(x=reorder(class, hwy, FUN=median), y=hwy)) +
  geom_boxplot() +
  coord_flip()
```

## The `geom_point` & `geom_smooth` functions {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=diamonds, mapping=aes(x=x, y=y)) +
  geom_point()
```

## The `geom_point` & `geom_smooth` functions {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg) +
  geom_point(mapping=aes(x=displ, y=hwy)) +
  facet_wrap(~class, nrow=2)
```

## The `geom_point` & `geom_smooth` functions {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg) +
  geom_point(mapping=aes(x=displ, y=hwy)) +
  facet_grid(drv~cyl)
```

## The `geom_point` & `geom_smooth` functions {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg) +
  geom_smooth(mapping=aes(x=displ, y=hwy), method = "loess")
```

## Combining geoms {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg) +
  geom_point(mapping=aes(x=displ, y=hwy)) +
  geom_smooth(mapping=aes(x=displ, y=hwy), method = "loess")
```

## Adding labels {.flexbox .vcenter}

```{r, eval=TRUE, echo=TRUE,fig.asp=0.5}
ggplot(data=mpg, aes(displ, hwy)) +
  geom_point(aes(color=class)) +
  geom_smooth(se=FALSE, method="loess") +
  labs(title = "Fuel efficiency generally decreases with engine size",
  subtitle = "Two seaters (sports cars) are an exception because of their light weight",
  caption = "Data from fueleconomy.gov")
```


## Themes

```{r, echo=FALSE, out.width='80%', fig.asp=.75, fig.align='center'}
knitr::include_graphics("images/images_4a.018.jpeg")
```

## Arranging Multiple Figures- Flexdashboard

- Modify YAML header to specify graph orientation

```{r, echo=TRUE, eval=FALSE}
---
title: Flexdashboard Options"
output: 
  flexdashboard::flex_dashboard:
    
   vertical_layout: 'fill' or 'scroll'
         or
     orientation: 'rows'
---
```
- Or specify data width to reorganize (try this in template)
https://rmarkdown.rstudio.com/flexdashboard/

## Arranging Multiple Figures- Flexdashboard

```{r, echo=FALSE, out.width='30%'}
knitr::include_graphics("images/felxdash_scroll.png")
knitr::include_graphics("images/dashboard_orient_rows.png")
knitr::include_graphics("images/dashboard_spec_col_widths.png")
```

## FINAL R INTERLUDE | Complete Exercise 2.5 {.flexbox .vcenter}
